flutter/lib/src/event_processor/android_platform_exception_event_processor.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+import '../../sentry_flutter.dart';
+import '../jvm/jvm_exception.dart';
+import '../jvm/jvm_frame.dart';
+
+/// Transforms an Android PlatformException to a human readable SentryException
+class AndroidPlatformExceptionEventProcessor implements EventProcessor {
+  const AndroidPlatformExceptionEventProcessor(this._options);
+
+  final SentryFlutterOptions _options;
+
+  // Because of obfuscation, we need to dynamically get the name
+  static final _platformExceptionType = (PlatformException).toString();
+
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {dynamic hint}) async {
+    final plaformException = event.throwable;
+    if (!(plaformException is PlatformException)) {
+      return event;
+    }
+
+    final nativeStackTrace = plaformException.stacktrace;
+    if (nativeStackTrace == null) {
+      return event;
+    }
+
+    try {
+      // PackageInfo has an internal cache, so no need to do it ourselves.
+      final packageInfo = await PackageInfo.fromPlatform();
+      return _processPlatformException(
+        event,
+        plaformException,
+        nativeStackTrace,
+        packageInfo.packageName,
+      );
+    } catch (e, stackTrace) {
+      _options.logger(
+        SentryLevel.info,
+        "Couldn't prettify PlatformException. "
+        'The exception will still be reported.',
+        exception: e,
+        stackTrace: stackTrace,
+      );
+      return event;
+    }
+  }
+
+  SentryEvent _processPlatformException(
+    SentryEvent event,
+    PlatformException exception,
+    String nativeStackTrace,
+    String packageName,
+  ) {
+    final jvmException =
+        _JvmExceptionFactory(packageName).fromJvmStackTrace(nativeStackTrace);
+
+    final exceptions = _removePlatformExceptionStackTraceFromValue(
+      event.exceptions,
+      exception,
+    );
+
+    return event.copyWith(
+      exceptions: [
+        ...?exceptions,
+        ...jvmException,
+      ],
+    );
+  }
+
+  /// Remove the StackTrace from [PlatformException] so the message on Sentry
+  /// looks much better.
+  List<SentryException>? _removePlatformExceptionStackTraceFromValue(
+    List<SentryException>? exceptions,
+    PlatformException platformException,
+  ) {
+    if (exceptions == null || exceptions.isEmpty) {
+      return null;
+    }
+    final exceptionCopy = List<SentryException>.from(exceptions);
+
+    final sentryExceptions = exceptionCopy
+        .where((element) => element.type == _platformExceptionType);
+    if (sentryExceptions.isEmpty) {
+      return [];
```

Comment:
```suggestion
      return null;
```

Should we just return null? since the method allows so, no need of allocating the empty list.